code-challenges.test.js

```diff
@@ -10,37 +18,198 @@
+
+  })
+})
+
+//  After running yarn jest, I should expect a good fail prompt saying the function is not defined
+//      ReferenceError: onlyRemsOf3 is not defined
 
 // b) Create the function that makes the test pass.
 
+//Create the function called onlyRemsOf3 with params of array of mixed data
+const onlyRemsOf3 = (mixedArr) => {
+  //An empty arr is created to store our final result until we call to return it
+  let result = []
+
+  //Iterate param of mixedArr with .filter() method w/ the value = data
+  mixedArr.filter(data => {
+    //Using the Number.isSafeInteger w/ data, we will determine if the data is an integer data type that is divisible by 3 w/ a remainder.
+    if (Number.isSafeInteger(data) % 3) {
+      //console.log(data % 3) ==>   ex1: --> [2, 0, -1, 0]    ex:2 --> [2, 1, -1]
+      
+      //If true we will push the data modulo by 3 into the empty arr of result
+      result.push(data % 3)
+    }
+  })
+  //After all the logic is done we will return to grab the expected output
+  return result
+}
+
+// onlyRemsOf3
+//     ✓ Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.167 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 0.91s.
+
+//====================================
+
+
+
+//    PROBLEM #3
+
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+//First create a Test function that takes in an array of nums data and returns sum of all nums cubed
+describe('sumCubed', () => {
+  
+  //Create an it function that will explain what is the purpose of the 'sumCubed' function
+  it('Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.', () => {
+
+    //Given 1st testCase that is array of nums with an Expected output: 99
+    const cubeAndSum1 = [2, 3, 4]
+    //if we invoke the function w/ cubeAndSum1 as the argument, expect it to equal --> given output
+    expect(sumCubed(cubeAndSum1)).toEqual(99)
+    
+    //Given 2nd testCase that is array of nums with an Expected output: 1125
+    const cubeAndSum2 = [0, 5, 10]
+    //if we invoke the function w/ cubeAndSum2 as the argument, expect it to equal --> given output
+    expect(sumCubed(cubeAndSum2)).toEqual(1125)
+
+  })
+})
+
+//  After running yarn jest, I should expect a good fail prompt saying the function is not defined
+//      ReferenceError: sumCubed is not defined
 
 // b) Create the function that makes the test pass.
+
+//Create the function called sumCubed with params of array of nums
+const sumCubed = (arrOfNums) => {  
```

Comment:
Super excited to see you using higher order functions on all of these problems!!